synthesizer/benches/utilities/workload.rs

```diff
@@ -0,0 +1,182 @@
+        // Check that the relevant blocks are stored in the object store.
+        all_data_is_stored &= match self.object_store.get("num_blocks") {
+            Err(err) => false,
+            Ok(num_blocks) => {
+                let num_blocks: u64 = num_blocks;
+                (0..num_blocks).all(|i| self.object_store.contains(format!("block_{}", i)))
+            }
+        };
+        // Check that the benchmark transactions are stored in the object store.
+        for benchmark in &self.benchmarks {
+            all_data_is_stored &= self.object_store.contains(benchmark.name())
+        }
+
+        // If all of the required items are not stored, clear the object store and initialize them.
+        if !all_data_is_stored {
+            println!("Data is not stored. Initializing...");
+            let mut setup_operations = vec![];
+            let mut benchmark_operations = vec![];
+
+            // Collect the operations for each benchmark.
+            for benchmark in &mut self.benchmarks {
+                let setup_ops = benchmark.setup_operations();
+                let benchmark_ops = benchmark.benchmark_operations();
+                setup_operations.push(setup_ops);
+                benchmark_operations.push((benchmark.name(), benchmark_ops));
+            }
+
+            // Fold the batches of setup operations across all benchmarks into a single sequence of batches.
+            let max_num_batches = setup_operations.iter().map(|operations| operations.len()).max().unwrap_or(0);
+            let mut aggregated_setup_operations = iter::repeat_with(Vec::new).take(max_num_batches).collect_vec();
+            for setup_ops in setup_operations {
+                for (i, operations) in setup_ops.into_iter().enumerate() {
+                    aggregated_setup_operations[i].extend(operations);
+                }
+            }
+
+            // Clear the object store.
+            self.object_store.clear().unwrap();
+            // Initialize the object store.
+            initialize_object_store(&mut self.object_store, aggregated_setup_operations, benchmark_operations)
+        } else {
+            println!("Data is stored. Loading...");
+            // Otherwise, load the items for the object store, initialize the VM, and return the VM, benchmark transactions, and rng.
+            // Initialize the RNG.
+            let mut rng = TestRng::fixed(self.object_store.get("seed").unwrap());
+
+            // Sample the private key.
+            let private_key = PrivateKey::<Testnet3>::new(&mut rng).unwrap();
+
+            // Initialize the VM.
+            let (vm, _) = initialize_vm::<C, _>(&private_key, &mut rng);
+
+            // Load the blocks.
+            let num_blocks: u64 = self.object_store.get("num_blocks").unwrap();
+            let blocks: Vec<Block<Testnet3>> =
+                (0..num_blocks).map(|i| self.object_store.get(&format!("block_{}", i)).unwrap()).collect_vec();
```

Comment:
```suggestion
            let blocks = (0..num_blocks).map(|i| self.object_store.get(&format!("block_{}", i)).unwrap());
```
there's no need to collect here